File: 프로그래머스/0/181921. 배열 만들기 2/배열 만들기 2.java
import java.util.List;
import java.util.ArrayList;

class Solution {
    public int[] solution(int l, int r) {
        List<Integer> list = new ArrayList<>();
        for (int i=1; Integer.parseInt(Integer.toBinaryString(i)) * 5 <= r; i++) {
            int value = Integer.parseInt(Integer.toBinaryString(i)) * 5;
            if (value >= l) list.add(value);
        }
        int[] answer = (list.size() > 0) ? new int[list.size()] : new int[]{-1};
        for (int i=0; i<list.size(); i++) {
            answer[i] = list.get(i).intValue();
        }
        return answer;
    }
}

File: 프로그래머스/0/181921. 배열 만들기 2/배열 만들기 2.py
def solution(l, r):
    answer = []
    i = 1
    while int(bin(i)[2:]) * 5 <= r:
        if int(bin(i)[2:]) * 5 >= l:
            answer.append(int(bin(i)[2:]) * 5)
        i += 1
    if not answer:
        answer.append(-1)
    return answer